Source/WebCore/platform/graphics/cg/NativeImageCG.cpp

```diff
@@ -69,6 +70,39 @@ DestinationColorSpace NativeImage::colorSpace() const
     return DestinationColorSpace(CGImageGetColorSpace(m_platformImage.get()));
 }
 
+static bool isHDRNativeImage(NativeImage& image)
+{
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+    return CGColorSpaceIsHDR(CGImageGetColorSpace(image.platformImage().get()));
+#pragma clang diagnostic pop
+}
+
+void NativeImage::draw(GraphicsContext& context, const FloatSize& imageSize, const FloatRect& destinationRect, const FloatRect& sourceRect, const ImagePaintingOptions& options)
+{
+    if (sourceRect.isEmpty() || !isHDRNativeImage(*this)) {
+        context.drawNativeImageInternal(*this, imageSize, destinationRect, sourceRect, options);
+        return;
+    }
+
+    // Create a temporary ImageBuffer for destinationRect with the current scaleFator.
+    auto imageBuffer = context.createScaledImageBuffer(destinationRect, context.scaleFactor(), DestinationColorSpace::SRGB(), RenderingMode::Unaccelerated, RenderingMethod::Local);
```

Comment:
This buffer should either use the image's colorspace, or the destination colorspace, but not sRGB.

Reply:
Fixed. DisplayP3 is used for iOS. Otherwise we match the destination context colorspace.